contracts/SpokePool.sol

```diff
@@ -1093,10 +1144,125 @@ abstract contract SpokePool is
+        // within the configured buffer. The owner should pause deposits/fills if this is undesirable.
+        // This will underflow if quoteTimestamp is more than depositQuoteTimeBuffer;
+        // this is safe but will throw an unintuitive error.
+
+        // slither-disable-next-line timestamp
+        uint256 currentTime = getCurrentTime();
+        if (currentTime - quoteTimestamp > depositQuoteTimeBuffer) revert InvalidQuoteTimestamp();
+
+        // fillDeadline is relative to the destination chain.
+        // Don’t allow fillDeadline to be more than several bundles into the future.
+        // This limits the maximum required lookback for dataworker and relayer instances.
+        // Also, don't allow fillDeadline to be in the past. This poses a potential UX issue if the destination
+        // chain time keeping and this chain's time keeping are out of sync but is not really a practical hurdle
+        // unless they are significantly out of sync or the depositor is setting very short fill deadlines. This latter
+        // situation won't be a problem for honest users.
+        if (fillDeadline < currentTime || fillDeadline > currentTime + fillDeadlineBuffer) revert InvalidFillDeadline();
+
+        // There are three cases for setting the exclusivity deadline using the exclusivity parameter:
+        // 1. If this parameter is 0, then there is no exclusivity period and emit 0 for the deadline. This
+        //    means that fillers of this deposit do not have to worry about the block.timestamp of this event changing
+        //    due to re-orgs when filling this deposit.
+        // 2. If the exclusivity parameter is less than or equal to MAX_EXCLUSIVITY_PERIOD_SECONDS, then the exclusivity
+        //    deadline is set to the block.timestamp of this event plus the exclusivity parameter. This means that the
+        //    filler of this deposit assumes re-org risk when filling this deposit because the block.timestamp of this
+        //    event affects the exclusivity deadline.
+        // 3. Otherwise, interpret this parameter as a timestamp and emit it as the exclusivity deadline. This means
+        //    that the filler of this deposit will not assume re-org risk related to the block.timestamp of this
+        //    event changing.
+        uint32 exclusivityDeadline;
+        if (exclusivityParameter == 0) {
+            exclusivityDeadline = 0;
+        } else {
+            if (exclusivityParameter <= MAX_EXCLUSIVITY_PERIOD_SECONDS) {
+                exclusivityDeadline = uint32(currentTime) + exclusivityParameter;
+            } else {
+                exclusivityDeadline = exclusivityParameter;
+            }
+
+            // As a safety measure, prevent caller from inadvertently locking funds during exclusivity period
+            //  by forcing them to specify an exclusive relayer.
+            if (exclusiveRelayer == address(0)) revert InvalidExclusiveRelayer();
+        }
```

Comment:
The following should be functionally equivalent, though it drops the subsequent revert possibility because it's hard to bundle that in cleanly (I think that check is _probably_ unnecessary anyway).

```suggestion
        uint32 exclusivityDeadline = exclusivityParameter;
        if (exclusivityParameter > 0 && exclusivityParameter <= MAX_EXCLUSIVITY_PERIOD_SECONDS) {
                exclusivityDeadline += uint32(currentTime);
        }
```

Or alternatively:

```suggestion
        uint32 exclusivityDeadline = exclusivityParameter;
        if (exclusivityParameter > 0) {
            if (exclusivityParameter <= MAX_EXCLUSIVITY_PERIOD_SECONDS) {
                exclusivityDeadline += uint32(currentTime);
            }
            
            // As a safety measure, prevent caller from inadvertently locking funds during exclusivity period
            //  by forcing them to specify an exclusive relayer.
            if (exclusiveRelayer == address(0)) revert InvalidExclusiveRelayer();
        }
```